challengeMk2/DataBase/ILocalDataService.cs

```diff
@@ -1,20 +1,21 @@
 using System;
 using System.Collections.Generic;
+using System.Threading.Tasks;
 using ChallengeMk2.Models;
 
 namespace ChallengeMk2.DataBase
 {
     public interface ILocalDataService
     {
-        void Initialize();
+        Task Initialize();
 
-        List<StarSystemDbItem> GetFullDb();
+        Task<List<StarSystemDbItem>> GetFullDb();
 
-        StarSystemDbItem GetItem(string name);
-        StarSystemDbItem GetItem(int id);
+        Task<StarSystemDbItem> GetItem(string name);
+        Task<StarSystemDbItem> GetItem(int id);
 
-        void SaveItem(StarSystemDbItem starSystem);
+        Task SaveItem(StarSystemDbItem starSystem);
 
-        void ClearDb();
+        Task ClearDb();

```

Comment:
This seems like a very destructive operation. From an API design perspective, I would not have this method in the same `ILocalDataService` interface as it's too easy to shoot yourself in the foot. What is the use case for this method ?

Reply:
it is used when the app updates datas from API after x days. 
Database is cleared before retrieving new datas : 
public async Task ClearDbAsync()
        {
            await database.DropTableAsync<StarSystemDbItem>();
            await database.CreateTableAsync<StarSystemDbItem>();
        }
Maybe, instead of dropping existing table before get and save datas , 
i can check:
 if (system is already in db) => update it
else (insert it).
I thought drop and re create table was less expensive than update it.
=> I will make this changes in a new branch, using a bulk insert/update items, instead of insert/update them one by one.

Reply:
Or can i add Bulk insert/update in this branch already ?

Reply:
EDIT : oki , definitively make a new branch ;)